std/algorithm/comparison.d

```diff
@@ -1298,6 +1298,62 @@ MaxType!T max(T...)(T args)
     return cast(typeof(return)) (chooseB ? b : a);
 }
 
+/**
+Iterates the passed range and selects the lowest value matching $(D pred).
+Can use a custom key getter, as defined by $(D keyGetter)
+(which is the identity by default).
+
+Params:
+    r = Range from which the minimum will be selected
+
+Returns: The lowest value according to $(D pred) of the passed-in values.
+*/
+auto selectOne(alias pred, alias keyGetter = (a) => a, Range)(Range r)
```

Comment:
We don't use the key getter style of programming most anywhere else in Phobos (though it does have appeal). For projections we use `map`. Here, custom `less` predicate would be used.